pkg/providers/amifamily/al2023.go

```diff
@@ -37,19 +37,18 @@ type AL2023 struct {
 
 func (a AL2023) DescribeImageQuery(ctx context.Context, ssmProvider ssm.Provider, k8sVersion string, amiVersion string) (DescribeImageQuery, error) {
 	ids := map[string]Variant{}
-	for arch, variants := range map[string][]Variant{
-		"x86_64": []Variant{VariantStandard, VariantNvidia, VariantNeuron},
-		"arm64":  []Variant{VariantStandard},
-	} {
-		for _, variant := range variants {
-			path := a.resolvePath(arch, string(variant), k8sVersion, amiVersion)
-			imageID, err := ssmProvider.Get(ctx, path)
-			if err != nil {
-				continue
+	for parameter, archToVariantMap := range a.parametersToVariantAndArchitecture(k8sVersion, amiVersion) {
```

Comment:
Just curious: Why is this so much more complicated for AL2023

Reply:
I'm realizing now it doesn't need to be, I tried to keep the arch but it was really only required for path resolution since we discover the architecture when describing the AMI. This could be simplified, but I'm reverting it anyway with the change to an invalidation controller.